### PR TITLE
TIP-733: Add form extensions for product grid header components

### PR DESCRIPTION
**Pull request by tamarasaurus**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

See https://github.com/akeneo/pim-community-dev/pull/6524 for the main description

This PR covers the following components in the product grid header: 
- [x] Title
- [x] Locale switcher
- [x] View selector
- [x] Create view
- [x] Create product 

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/locale-switcher.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                this.config = config.config;
+
+                return BaseForm.prototype.initialize.apply(this, arguments);
+            },
+
+            /**
+             * {@inheritdoc}
+             */
+            render() {
+                this.fetchLocales().then((locales) => {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/locale-switcher.js]: the locale list fetch should be done in the configure.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/view-selector.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    ) {
+        return BaseForm.extend({
+            id: 'view-selector',
+            className: 'pull-right',
+
+            /**
+             * {@inheritdoc}
+             */
+            render() {
+                FetcherRegistry.initialize().done(function () {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/view-selector.js]: no need to initialize the fetcher registry anymore

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/view-selector.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                    FormBuilder.buildForm('pim-grid-view-selector').then(function (form) {
+                        return form.configure('product-grid').then(function () {
+                            form.setElement('#view-selector').render();
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/view-selector.js]: We should avoid using id IMHO

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/grid/locale-switcher.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        <%= i18n.getFlag(currentLocaleCode, false) %>&nbsp;<%- getDisplayName(currentLocaleCode) %>
+        <div class="AknDropdownButton-caretContainer">
+            <i class="AknCaret AknCaret--big"></i>
+        </div>
+    </div>
+    <ul class="AknDropdown-menu">
+        <% _.each(locales, (locale) => { %>
+             <li>
+                <a class="AknDropdown-menuLink <% if (currentLocaleCode == locale.code) { %>AknDropdown-menuLink--active<% } %>"
+                    href="#<%- generateUrl(locale.code) %>"
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/grid/locale-switcher.html]: We should not use href anymore

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/grid/locale-switcher.html [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    <div class="AknDropdownButton-button" data-toggle="dropdown">
+        <%= i18n.getFlag(currentLocaleCode, false) %>&nbsp;<%- getDisplayName(currentLocaleCode) %>
+        <div class="AknDropdownButton-caretContainer">
+            <i class="AknCaret AknCaret--big"></i>
+        </div>
+    </div>
+    <ul class="AknDropdown-menu">
+        <% _.each(locales, (locale) => { %>
+             <li>
+                <a class="AknDropdown-menuLink <% if (currentLocaleCode == locale.code) { %>AknDropdown-menuLink--active<% } %>"
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/grid/locale-switcher.html]: we should not use <a> tags

Comment (tamarasaurus) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/grid/locale-switcher.html]: do you mean <a> tags?

**Pull request merged**